.circleci/config.yml

```diff
@@ -0,0 +1,66 @@
+version: 2.1
+
+orbs:
+  win: circleci/windows@2.4.0
+
+jobs:
+  build:
+    executor: win/default
+    
+    environment:
+      THIRDPARTY_HOME: .\thirdparty
+      SNAPPY_HOME: ${THIRDPARTY_HOME}\snappy-1.1.7
+      SNAPPY_INCLUDE: ${SNAPPY_HOME};${SNAPPY_HOME}\build
+      SNAPPY_LIB_DEBUG: ${SNAPPY_HOME}\build\Debug\snappy.lib
+      SNAPPY_LIB_RELEASE: ${SNAPPY_HOME}\build\Release\snappy.lib
+      LZ4_HOME: ${THIRDPARTY_HOME}\lz4-1.8.3
+      LZ4_INCLUDE: ${LZ4_HOME}\lib
+      LZ4_LIB_DEBUG: ${LZ4_HOME}\visual\VS2010\bin\x64_Debug\liblz4_static.lib
+      LZ4_LIB_RELEASE: ${LZ4_HOME}\visual\VS2010\bin\x64_Release\liblz4_static.lib
+      ZSTD_HOME: ${THIRDPARTY_HOME}\zstd-1.4.0
+      ZSTD_INCLUDE: ${ZSTD_HOME}\lib;${ZSTD_HOME}\lib\dictBuilder
+      ZSTD_LIB_DEBUG: ${ZSTD_HOME}\build\VS2010\bin\x64_Debug\libzstd_static.lib
+      ZSTD_LIB_RELEASE: ${ZSTD_HOME}\build\VS2010\bin\x64_Release\libzstd_static.lib
+      CMAKE_GENERATOR: Visual Studio 19 Win64
+      DEV_ENV: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\devenv.com
+
+    steps:
+      - checkout
+      - run:
+          name: "Install thirdparty dependencies"
+          command: |
+            md %THIRDPARTY_HOME%
+            echo "Building Snappy dependency..."
+            cd %THIRDPARTY_HOME%
+            curl --fail --silent --show-error --output snappy-1.1.7.zip --location https://github.com/google/snappy/archive/1.1.7.zip
+            unzip snappy-1.1.7.zip
+            cd snappy-1.1.7
+            mkdir build
+            cd build
+            cmake -G "%CMAKE_GENERATOR%" ..
+            msbuild Snappy.sln /p:Configuration=Debug /p:Platform=x64
+            msbuild Snappy.sln /p:Configuration=Release /p:Platform=x64
+            echo "Building LZ4 dependency..."
+            cd %THIRDPARTY_HOME%
+            curl --fail --silent --show-error --output lz4-1.8.3.zip --location https://github.com/lz4/lz4/archive/v1.8.3.zip
+            unzip lz4-1.8.3.zip
+            cd lz4-1.8.3\visual\VS2010
+            ps: $CMD="$Env:DEV_ENV"; & $CMD lz4.sln /upgrade
+            msbuild lz4.sln /p:Configuration=Debug /p:Platform=x64
+            msbuild lz4.sln /p:Configuration=Release /p:Platform=x64
+            echo "Building ZStd dependency..."
+            cd %THIRDPARTY_HOME%
+            curl --fail --silent --show-error --output zstd-1.4.0.zip --location https://github.com/facebook/zstd/archive/v1.4.0.zip
+            unzip zstd-1.4.0.zip
+            cd zstd-1.4.0\build\VS2010
+            ps: $CMD="$Env:DEV_ENV"; & $CMD zstd.sln /upgrade
+            msbuild zstd.sln /p:Configuration=Debug /p:Platform=x64
+            msbuild zstd.sln /p:Configuration=Release /p:Platform=x64
+      - run:
+          name: "Build and test"
+          command: |
+            md .\build
+            cd .\build
+            cmake -G "%CMAKE_GENERATOR%" -DCMAKE_BUILD_TYPE=Debug -DOPTDBG=1 -DPORTABLE=1 -DSNAPPY=1 -DLZ4=1 -DZSTD=1 -DXPRESS=1 -DJNI=1 ..
```

Comment:
Do we need to do actual build here?

Reply:
I followed the existing commands in appveyor's config.
I think this just generates the build files for windows instead of actually doing the build.

Reply:
appveyor has a step:
```
build:
  project: build\rocksdb.sln
```
I don't know what it means. Maybe it automatically calls msbuild using the file. It doesn't seem to be the case here.

Reply:
Indeed the CMake step generates the VS projects build files